neutrino-custom/index.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+const { join } = require('path');
+const { ProvidePlugin } = require('webpack');
+const lint = require('./lint');
+const development = require('./development');
+const production = require('./production');
+const test = require('./test');
+const notTest = require('./not-test');
+const localWatch = require('./local-watch');
+const react = require('neutrino-preset-react');
+const HtmlPlugin = require('html-webpack-plugin');
+
+const HTML_MINIFY_OPTIONS = {
+    useShortDoctype: true,
+    keepClosingSlash: true,
+    collapseWhitespace: true,
+    preserveLineBreaks: true
+};
+
+const entries = ['index', 'admin', 'perf', 'logviewer', 'failureviewer', 'userguide'];
+
+module.exports = neutrino => {
+    const { source } = neutrino.options;
+    const { NODE_ENV, LOCAL_WATCH } = process.env;
+
+    neutrino.config
+        // Environment-specific configuration
+        .when(NODE_ENV === 'development' && !LOCAL_WATCH, () => neutrino.use(development))
+        .when(NODE_ENV === 'production' || LOCAL_WATCH || NODE_ENV === 'test', () => neutrino.use(production))
+        .when(NODE_ENV === 'test' && !LOCAL_WATCH, () => neutrino.use(test), () => neutrino.use(notTest))
+        .when(LOCAL_WATCH, () => neutrino.use(localWatch))
+
+        // Common configuration
+        .plugins
+            .delete('html').end()
+        .plugin('provide')
+            // Now provide a few globals (providing angular here does not seem to work, though)
+            .use(ProvidePlugin, [{
+                $: require.resolve('jquery'),
+                jQuery: require.resolve('jquery'),
+                'window.$': require.resolve('jquery'),
+                'window.jQuery': require.resolve('jquery'),
+                React: require.resolve('react'),
+                _: require.resolve('lodash'),
+                treeherder: require.resolve(join(source, 'js/treeherder.js')),
+                treeherderApp: require.resolve(join(source, 'js/treeherder_app.js')),
+                perf: require.resolve(join(source, 'js/perf.js')),
+                admin: require.resolve(join(source, 'js/admin.js')),
+                failureViewerApp: require.resolve(join(source, 'js/failureviewer.js')),
+                logViewerApp: require.resolve(join(source, 'js/logviewer.js')),
+                userguideApp: require.resolve(join(source, 'js/userguide.js'))
+            }]).end()
+        .entry('index')
+            .delete(require.resolve('babel-polyfill')).end()
```

Comment:
Neutrino v5 added babel-polyfill by default to the index entry point. Treeherder doesn't use this right now, so I removed it to avoid breaking anything on that entry. I will comment.